jerry-core/parser/js/js-parser-statm.c

```diff
@@ -710,9 +710,8 @@ parser_parse_function_statement (parser_context_t *context_p) /**< context */
                 && context_p->token.lit_location.type == LEXER_IDENT_LITERAL);
 
 #if ENABLED (JERRY_ES2015)
-  if (context_p->next_scanner_info_p->source_p == context_p->source_p)
```

Comment:
This is the way it should work. If you check the type, but not the `source_p`, it might refer to something detected later in the source code.